pkg/upgrade/upgrade.go

```diff
@@ -204,7 +243,16 @@ func upgradeToVer3(cli *clientv3.Client, uctx Context) error {
 
 	ops := make([]clientv3.Op, 0, len(etcdKeyUpgrades))
 	for _, pair := range etcdKeyUpgrades {
-		resp, err := cli.Get(uctx.Context, pair.old.Path(), clientv3.WithPrefix())
+		resp, err := cli.Get(ctx, pair.new.Path(), clientv3.WithPrefix())
+		if err != nil {
+			return err
+		}
+		if len(resp.Kvs) > 0 {
```

Comment:
what's the benefits of we changing the upgrading detections? If user removes source configs after scheduler.Start() and DM-master failed to save current version in etcd (may caused by we write a buggy upgradeToVerX in future), the new leader will add source configs again